# Tipos de variables en Python

In [2]:
3.0 #números
""" Este es un comentario 
mas largo 
y necesitamos abrir y 
cerrar el fin del comentario """

3.0

In [4]:
x = 3 #asignación de un valor de una variable 


In [5]:
print(x)

3


In [7]:
y=type(x)

In [24]:
str(5/2)

'2.5'

In [5]:
x = 3.0

In [6]:
x

3.0

In [25]:
type(x) #el tipo de variable cambió debido al contenido que se le asignó

int

In [42]:
variable = 5

In [43]:
variable

5

In [44]:
variable + 5


10

In [45]:
variable = variable + 5
print(variable) #print devuelve en pantalla el valor dentro del los ()

10


In [46]:
variable = x
print(type(x))

<class 'str'>


In [52]:
x = "string" #las comillas nos indican que es una cadena de caracteres

In [50]:
x

'string'

In [48]:
type(x)

str

In [58]:
q=''

In [59]:
x=q

In [35]:
variable = "x"

In [41]:
variable

10

In [37]:
print(variable)

x


In [38]:
print(x)

string


In [39]:
variable = 10

In [40]:
type(variable)

int

In [23]:
variable = 8**6
variable

262144

In [24]:
8**6

262144

----

# Operaciones con variables

Con las variables que ya estudiamos, podemos definir operaciones.  

**Suma** 

In [1]:
3 + 7

10

De igual manera, podemos realizar la suma con dos variables definidas, es decir

In [1]:
x = 3
y = 7

In [2]:
x + y

10

Podemos sumar:

* `float` con `float` 
* `float` con `int`
* `str` con `str`

`float` con `float`

In [60]:
x = 1.2314
y = 3.1415
x + y 

4.3729000000000005

Nos regresará un `float`

In [61]:
type(x+y)

float

`float` con `int`

In [62]:
x = 1.2314
y = 3
x + y 

4.2314

Nos regresará un `float`

In [63]:
type(x+y)

float

`str` con `str`

In [69]:
x = 'Buenas'    # 'Buenas'+'Tardes'
y = 'Tardes'    # 'Buenas '+'Tardes'
x +' '+ y       # 'Buenas'+' '+'Tardes'

'Buenas Tardes'

Nos regresará un `str`

In [70]:
type(x+y)

str

---
**Sustracción**

In [71]:
3 - 7

-4

In [72]:
x = 10 
y = 5.5
x - y

4.5

Podemos restar:

* `float` con `float` 
* `float` con `int`

---
**Multiplicación**

In [2]:
3 * 7

21

In [3]:
x = 9 
y = 5.3
x * y

47.699999999999996

Aquí podemos observar algo **interesante**. Si hacemos la multiplicación a "mano" $9 \times 5.3 = 47.7$ ¿Qué pasa aquí? A esto se le conoce como _**Error de redondeo**_, este es ocasionado por Python a la hora de almacenar los números decimales en binario. Existen bibliotecas que nos evitan este problema, pero en general, todos los lenguajes de programación tienen este problema. 

Podemos mutiplicar:

* `float` con `float` 
* `float` con `int`

---
**División**

In [4]:
10/4

2.5

In [5]:
x = 20 
y = 13.2
x / y

1.5151515151515151

Podemos dividir:

* `float` con `float` 
* `float` con `int`

Como podemos esperar, dividir entre cero nos genera error:

In [73]:
5 / 0

ZeroDivisionError: division by zero

Python respeta la jerarquía de operaciones, en caso de combinar operaciones también se puede hacer uso de los parentesis.

---
**Potencia**

In [9]:
2.5**4 #Elevar 2.5 a la 4

39.0625

In [10]:
9**0.5 #Raíz cuadrada

3.0

In [84]:
compleja=(-1)**(0.5)#Existen también numeros complejos en Python

In [88]:
compleja.imag

1.0

In [14]:
10**-4 

0.0001

In [18]:
x = 3
y = 3.2
x**y

33.63473536961897

Podemos potenciar:

* `float` con `float` 
* `float` con `int`

---
**Módulo**

Es el también conocido como resto de la división. 

Tenemos que _dividendo = divisor*cociente + resto_

Cuando la división es exacta, el módulo es cero. 

In [89]:
4 % 2 #Si dividimos 4 entre 2, da 2, no sobra y por tanto el módulo es 0.

0

In [90]:
7 % 3 #Si dividimos 7 entre 3, da 2 y nos sobra 1, por lo que el módulo es 1.

1

In [91]:
x = 10 
y = 5.5
x % y

4.5

La herramienta del módulo nos puede ayudar, por ejemplo, a encontrar números pares o multiplos de otros números. 